vllm/v1/attention/backends/mla/common.py

```diff
@@ -436,6 +436,26 @@ class MLACommonMetadataBuilder(AttentionMetadataBuilder[M]):
     NOTE: Please read the comment at the top of the file before trying to
     understand this class
     """
+
+    # Whether the backend supports reordering the batch such that
+    # short sequences (i.e. verification for speculative decoding) are
+    # classified as decode requests.
+    # If True, this will increase `reorder_batch_threshold` (below) when
+    # speculative decoding is enabled.
+    supports_spec_as_decode: ClassVar[bool] = False
```

Comment:
Should we throw an error when a user tries to enable spec decode with cutlass-mla and specify use `VLLM_ATTENTION_BACKEND=FLASHINFER_MLA` instead?

Reply:
In the default case, I feel like it _should_ work for any backend, and this flag only declares that the kernel is opting in to batch reordering such that verification requests with q_len > 1 are classified as decodes. Without this, it should still run.

There is still an unexplained issue with CutlassMLA that we might want to warn about / default away from. But that doesn't seem as much to do with the settings here

Reply:
Sorry, I tagged the wrong line but yes my comment was to warn/error out in paths where users might see wrong results.

Reply:
Bug is fixed now, resolving the conversation.